io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/internal/simpletype/SimpleTypeUtil.java

```diff
@@ -119,6 +132,31 @@ else if (simpleTypeValue instanceof XmlDateTime) {
 			}
 		}
 
+		if (value instanceof BigDecimal) {
+			// Create a DecimalFormat object without scientific notation
+			DecimalFormat decimalFormat = new DecimalFormat("0.##############");
+			// Format the BigDecimal
+			return decimalFormat.format(value);
+		}
+		else
+		// Handle primitive types
+		if (value instanceof Double || value instanceof Float || value instanceof Long
+				|| value instanceof Integer) {
+			return String.valueOf(value);
+		}
+		else if (value instanceof XmlAnySimpleType) {
+			Number numberValue = (Number) value;
+			if (value instanceof XmlDouble || value instanceof XmlFloat || value instanceof XmlLong
+					|| value instanceof XmlInteger) {
+				// Return the string value of the number
+				return String.valueOf(numberValue);
+			}
+			else if (value instanceof XmlDecimal) {
+				BigDecimal decimalValue = ((XmlDecimal) value).getBigDecimalValue();
+				return decimalValue.stripTrailingZeros().toPlainString();
+			}
+		}
+
```

Comment:
The intention with the use of the `xmlbeans` simple type classes is to use them to [get the correct string representation](https://github.com/halestudio/hale/blob/c5a92100ad860826d107a13a0dd20084a727f948/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/internal/simpletype/SimpleTypeUtil.java#L115) to use in the XML.

You already extended the type map with the respective classes, and this is used [here](https://github.com/halestudio/hale/blob/c5a92100ad860826d107a13a0dd20084a727f948/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/internal/simpletype/SimpleTypeUtil.java#L82).
But instead of the custom handling that you added here now, what is missing are converters that convert the related type bindings to and from the XML simple type classes. Like for example these converters from `Date` to `XmlDateTime` and vice versa:

- [DateToXmlDateTime](https://github.com/halestudio/hale/blob/master/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/internal/simpletype/converters/DateToXmlDateTime.java)
- [XmlDateTimeToDate](https://github.com/halestudio/hale/blob/master/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/internal/simpletype/converters/XmlDateTimeToDate.java)

To make converters available there is an existing extension point to register the converters. See example [here](https://github.com/halestudio/hale/blob/c5a92100ad860826d107a13a0dd20084a727f948/io/plugins/eu.esdihumboldt.hale.io.gml/plugin.xml#L1495).